src/diffpy/pdfgui/control/fitdataset.py

```diff
@@ -785,10 +785,10 @@ def grid_interpolation(x0, y0, x1, youtleft=0.0, youtright=0.0):
 
     Return numpy.array of interpolated y1 values.
     """
-    x0 = numpy.array(x0, copy=False, dtype=float)
-    y0 = numpy.array(y0, copy=False, dtype=float)
+    x0 = numpy.asarray(x0, copy=None, dtype=float)
```

Comment:
Reference:

https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword